### DISCUSSION TITLE
How to validate array length

### DISCUSSION THREAD

**Discussion by BigDog1400** [gvergnaud/ts-pattern#181] [Q&A]
Hi. I am trying to understand how to make a case for 1) array.length is greater than > 0 2) array.length is 0.
![image](https://github.com/gvergnaud/ts-pattern/assets/63280663/e7da5f61-fddb-4aac-be40-2394f15b31b3)
This works fine. But i don't want to nest another match. I will like to something like this, but i think that it cause another permutations to exist and it throw some errors
![image](https://github.com/gvergnaud/ts-pattern/assets/63280663/c1f47efa-cfe1-49e3-93bd-6f97e784a1d4)
I also try with the P.array but i don't fully understand the [release notes](https://github.com/gvergnaud/ts-pattern/releases/tag/v5.0.0) 😅 . 

so... does someone knows the best way to validate array length?

**Comment by gvergnaud** [comment]
Hey! 
The best way to match a specific array length is to use a (variadic) tuple pattern: 

```ts
match(array)
  .with(
    { status: 'success', data: [] }
    () => "matches empty arrays"
  )
  .with(
    { status: 'success', data: [P.any] }
    () => "matches arrays with exactly 1 element"
  )
  .with(
    { status: 'success', data: [P.any, P.any] }
    () => "matches arrays with exactly 2 elements"
  )
  .with(
    { status: 'success', data: [P.any, ...P.array()] }
    () => "matches arrays with at least 1 element"
  )
```
